src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -299,10 +299,9 @@ static boolean canWalkOn(BlockStateInterface bsi, int x, int y, int z, IBlockSta
         if (block == Blocks.LADDER || (block == Blocks.VINE && Baritone.settings().allowVines.value)) { // TODO reconsider this
             return true;
         }
-        if (block == Blocks.FARMLAND || block == Blocks.GRASS_PATH) {
-            return true;
-        }
-        if (block == Blocks.ENDER_CHEST || block == Blocks.CHEST || block == Blocks.TRAPPED_CHEST) {
+        if (block == Blocks.FARMLAND || block == Blocks.GRASS_PATH ||
+            block == Blocks.ENDER_CHEST || block == Blocks.CHEST || block == Blocks.TRAPPED_CHEST ||
+            block == Blocks.END_PORTAL_FRAME || block == Blocks.ENCHANTING_TABLE) {
```

Comment:
Why end portal frame and enchanting table?

Reply:
I asked that already, it's in the original list somewhere??

Reply:
blocks that are non-full appear to need to be in `canWalkOn` for baritone to be able to walk on them. If I'm mistaken about the purpose of this function, this change isn't actually necessary for the other stuff to work.